src/main/kotlin/nexters/linkllet/common/support/LoginUserResolver.kt

```diff
@@ -24,7 +25,16 @@ class LoginUserResolver(
             binderFactory: WebDataBinderFactory?,
     ): String {
         val request = webRequest.nativeRequest as HttpServletRequest
-        val token = AuthorizationExtractor.extractToken(request)
-        return jwtProvider.getPayload(token)
+
+        val deviceId = DeviceHeaderExtractor.extractDeviceId(request)
+        val jwtToken = AuthorizationExtractor.extractToken(request)
+
+        if (hasBothDeviceIdAndToken(deviceId, jwtToken)) throw UnauthorizedException()
+        else if (deviceId.isNullOrBlank()) return jwtProvider.getPayload(jwtToken)
+        else if (jwtToken.isNullOrBlank()) return deviceId
+        else throw UnauthorizedException()
```

Comment:
여담으로 이것도 return문을 앞에 쓸 수 있습니다 ㅎㅎ



```suggestion
return if (hasBothDeviceIdAndToken(deviceId, jwtToken)) throw UnauthorizedException()
        else if (deviceId.isNullOrBlank()) jwtProvider.getPayload(jwtToken)
        else if (jwtToken.isNullOrBlank()) deviceId
        else throw UnauthorizedException()
```

Reply:
오 일단 이게 가능하다는 거를 처음알게 됬습니다, 신기한 코틀린......
근데 이렇게 되면 return이 식 자체를 반환하는 고차함수가 되는 거 같아... 값만 반환하도록 하는 현 상태를 유지하고 싶어요!!